compiler/test/runnable/test24010.d

```diff
@@ -0,0 +1,60 @@
+// https://issues.dlang.org/show_bug.cgi?id=24010
+/* RUN_OUTPUT:
+---
+statement
+~this
+~this
+4
+val = 1
+destruct 1
+---
+*/
+
+alias AliasSeq(TList...) = TList;
+
+import core.stdc.stdio;
+
+int i = 0;
+struct A {
+    ~this() {
+        printf("~this\n");
+        i *= 2;
+    }
+}
+
+void test1() {
+    {
+        AliasSeq!(A, A) params;
+        printf("statement\n");
+        i = 1;
+    }
+    printf("%d\n", i);
+    assert(i == 4);
+}
+
+/***** from issue 17128 *****/
+
+struct Foo {
+    int val = 0;
+        ~this() {
+        printf("destruct %d\n", val);
+        assert(val == 1);
+    }
+}
+
+void bar(ARGS...)() {
+    ARGS args;
+    args[0].val = 1;
+    printf("val = %d\n", args[0].val);
+}
+
+void test2() {
+    bar!Foo();
+}
+/****************************/
+
+void main()
+{
+    test1();
+    test2();
```

Comment:
The tests in https://github.com/dlang/dmd/pull/15366/files#diff-d6cdd89ae270292efd2462b4f14ab839114f4ec71ddb37d52eb9d84e3238965bR4838
1. verify the correct order of the destructor calls
2. verify that the tuple name can also be used to access the fields
3. is integrated into existing sdtor.d tests, where the other destructor tests are, and merging into existing test files helps the test suite run faster

So, please replace this test with the other one.

Reply:
Done.